### PR TITLE
style: fix InputNumber align issue

### LINKED ISSUE #40936: InputNumber 在置于 alignItems:baseline 的容器内时，基线位置不对
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/inline-login-form-antd-5-2-2-forked-f4z9kz?file=/demo.tsx)

### Steps to reproduce

详见 codesandbox。将几种不同的组件放在一行内，设置父容器 `alignItems:baseline`，会发现 `InputNumber` 和其它组件不在一条线上。也可以将 `Space` 作为父容器复现。

这个问题最开始在使用 procomponents 时发现，`ProFormDigit` 和 `ProFormDigitRange` 也受此影响表现异常。在 antd@4 中也存在此问题。

### What is expected?

基线保持一致

### What is actually happening?

`InputNumber` 基线与其他组件不一致

| Environment | Info |
| --- | --- |
| antd | 5.2.2 |
| React | latest |
| System | macOS latest |
| Browser | Chrome latest |

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/40936

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix InputNumber cannot align well by baseline.        |
| 🇨🇳 Chinese | 修复 InputNumber 和其他组件无法基线对齐的问题。          |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 8c7174e</samp>

Fix input number alignment issue by removing redundant style rule in `components/input-number/style/index.ts`.

### <samp>🤖 Generated by Copilot at 8c7174e</samp>

* Removed redundant margin style from input number component to fix misalignment issue in some browsers ([link](https://github.com/ant-design/ant-design/pull/42580/files?diff=unified&w=0#diff-63d0bbc4ed8cb1745da77de552d75aa0fd581ef3eb2d25d0108b2d55eb5f1857L126))

### CODE REVIEW

File: components/input-number/style/index.ts [hunk 1]
```diff
@@ -123,7 +123,6 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
           '&-wrapper': {
             display: 'inline-block',
             textAlign: 'start',
-            verticalAlign: 'top', // https://github.com/ant-design/ant-design/issues/6403
```
Comment (afc163) [inline components/input-number/style/index.ts:126]: 看上去没啥用。

**Comment by afc163** [comment]
有点问题：https://app.argos-ci.com/ant-design/ant-design/builds/8606/48431825

File: components/input-number/style/index.ts [hunk 2]
```diff
@@ -124,7 +124,7 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
             display: 'inline-block',
             textAlign: 'start',
             verticalAlign: 'top', // https://github.com/ant-design/ant-design/issues/6403
-
+
```
Comment (afc163) [inline components/input-number/style/index.ts:127]:
```suggestion

```

File: components/input-number/style/index.ts [hunk 3]
```diff
@@ -157,7 +157,6 @@ const genInputNumberStyles: GenerateStyle<InputNumberToken> = (token: InputNumbe
             transition: `all ${motionDurationMid} linear`,
             appearance: 'textfield',
             fontSize: 'inherit',
-            verticalAlign: 'top',
```
Comment (afc163) [inline components/input-number/style/index.ts:160]: @zombieJ 这个是解决啥问题：https://github.com/ant-design/ant-design/commit/22193f70fa568049565e32deaf44025d7e183424#diff-084cf53900ef3e8965ef7327df045ce91e2e943919c27dc0c3595881569ba1a6R161

Comment (afc163) [reply components/input-number/style/index.ts:160]: 哦，这里有用。

图片

**Review by zombieJ — APPROVED** [review]

**Pull request merged**